File: year2_1819/computer_programming_3_algorithms_data_structures/2019_01_26_16_02_32_255813_organised/w3_sequences_sets_and_maps_review/phone_numbers.py

print('Enter a name and number, or a name and ? to query (!! to exit)')
phone_book = {}
command = input()
while command != '!!':
	name, arg = command.split()
	if arg == '?':
		if name in phone_book.keys():
			print('{} has number {}'.format(name,phone_book[name]))
		else:
			print('Sorry, there is no {}'.format(name))
	else:
		phone_book[name] = arg

	command = input()
print('Bye')

File: year2_1819/computer_programming_4_object_oriented_programming/2019_05_26_java_oop_organised/w6_java_collections/phone_numbers.java

import java.util.Scanner;
import java.util.Map;
import java.util.HashMap;

public class PhoneNumbers
{
    public static void main(String [] args)
    {
        System.out.println("Enter a name and number, or a name and ? to query (!! to exit)");
        Scanner in = new Scanner(System.in);
        Map<String, String> phonebook = new HashMap<>();
        
        String command = in.nextLine();
        while (!command.equals("!!")) {
            String[] parts = command.split(" ");
            String name = parts[0];
            String request = parts[1];
            if (request.equals("?")) {
                if (phonebook.containsKey(name)) 
                    System.out.printf("%s has number %s\n", name, phonebook.get(name));
                else 
                    System.out.printf("Sorry, there is no %s\n", name);
            }
            else 
                phonebook.put(name, request);
            command = in.nextLine();
        }
        System.out.println("Bye");
    }
}